### PR TITLE
feat(zk): add `extended_domain_` to `Prover`

### PR DESCRIPTION
**Pull request by TomTaehoonKim**
# Description

This PR adds a new member `extended_domain_` to `Prover`.

### CODE REVIEW

**Review by chokobole — APPROVED** [review]
LGTM

File: tachyon/zk/base/prover.h [hunk 1]
```diff
@@ -10,13 +10,14 @@
...
 
+#include "tachyon/math/polynomials/univariate/univariate_evaluation_domain.h"
 #include "tachyon/zk/base/blinded_polynomial.h"
 #include "tachyon/zk/base/blinder.h"
 #include "tachyon/zk/transcript/transcript.h"
 
 namespace tachyon::zk {
 
-template <typename _PCSTy>
+template <typename _PCSTy, size_t ExtendedMaxDegree>
```
Comment (dongchangYoo) [inline tachyon/zk/base/prover.h:20]: We agreed not to declare implementations like `UnivariateXXX` directly at the `Prover` level. Therefore, please consider changing from `size_t ExtendedMaxDegree` to `typename ExtendedDomain` in this part.
This is the content recently discussed with team lead.

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**